### PR TITLE
Fix bug match_anchor_to_bbox

### PR DESCRIPTION
**Pull request by AnirudhDagar**
*Description of changes:*
This PR fixes a small uncaught bug in scratch implementation as shown [here](https://discuss.d2l.ai/t/anchor-boxes/1603/4).
Also, add more explanation to code for `multibox_prior`.

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

**Comment by AnirudhDagar** [comment]
Closed and reopened pr to trigger CI.

File: chapter_computer-vision/ssd.md [hunk 1]
```diff
@@ -716,3 +716,4 @@ E. Refer to the SSD paper. What methods can be used to evaluate the precision of
 :begin_tab:`pytorch`
 [Discussions](https://discuss.d2l.ai/t/1604)
 :end_tab:
+
```
Comment (AnirudhDagar) [inline chapter_computer-vision/ssd.md:719]: Added this line to rerun CI for SSD with fixed code.

Comment (AnirudhDagar) [reply chapter_computer-vision/ssd.md:719]: Will remove later.

**Comment by mli** [comment]
Job d2l-en/PR-1630/1 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1630/

**Comment by mli** [comment]
Job d2l-en/PR-1630/2 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1630/

**Comment by mli** [comment]
Job d2l-en/PR-1630/3 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1630/

**Comment by mli** [comment]
Job d2l-en/PR-1630/4 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1630/

File: chapter_computer-vision/anchor.md [hunk 2]
```diff
@@ -275,9 +281,15 @@ def match_anchor_to_bbox(ground_truth, anchors, device, iou_threshold=0.5):
     box_j = indices[max_ious >= 0.5]
     anchors_bbox_map[anc_i] = box_j
     # Find the largest iou for each bbox
-    anc_i = np.argmax(jaccard, axis=0)
-    box_j = np.arange(num_gt_boxes)
-    anchors_bbox_map[anc_i] = box_j
+    col_discard = np.full((num_anchors,), -1)
+    row_discard = np.full((num_gt_boxes,), -1)
+    for _ in range(num_gt_boxes):
+      max_idx = np.argmax(jaccard)
```
Comment (astonzhang) [inline chapter_computer-vision/anchor.md:287]: Please ensure 4-space indentation throughout code

**Comment by mli** [comment]
Job d2l-en/PR-1630/6 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1630/

**Comment by mli** [comment]
Job d2l-en/PR-1630/7 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1630/

**Comment by astonzhang** [comment]
Thanks. Let's also try other ways to reduce the running time for SSD.

**Pull request merged**